File: core-service/src/main/java/com/sandjelkovic/dispatchd/domain/data/repository/EpisodeRepository.java
package com.sandjelkovic.dispatchd.domain.data.repository;


import com.sandjelkovic.dispatchd.domain.data.entity.Episode;
import com.sandjelkovic.dispatchd.domain.data.entity.Season;
import com.sandjelkovic.dispatchd.domain.data.entity.TvShow;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface EpisodeRepository extends CrudRepository<Episode, Long> {
	Page<Episode> findByTvShow(TvShow show, Pageable pageable);

	List<Episode> findByTvShow(TvShow show);

	Page<Episode> findBySeason(Season season, Pageable pageable);

	List<Episode> findBySeason(Season season);
}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/contentservice/data/repository/EpisodeRepository.kt
package com.sandjelkovic.dispatchd.contentservice.data.repository

import com.sandjelkovic.dispatchd.contentservice.data.entity.Episode
import com.sandjelkovic.dispatchd.contentservice.data.entity.Season
import com.sandjelkovic.dispatchd.contentservice.data.entity.Show
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository

/**
 * @author sandjelkovic
 * @date 27.1.18.
 */
interface EpisodeRepository : PagingAndSortingRepository<Episode, Long> {
    fun findByShow(show: Show, pageable: Pageable): Page<Episode>

    fun findByShow(show: Show): List<Episode>

    fun findBySeason(season: Season, pageable: Pageable): Page<Episode>

    fun findBySeason(season: Season): List<Episode>
}
